### DISCUSSION TITLE
[Request] Add a 'Delay_value" input to the delay component.

### DISCUSSION THREAD

**Discussion by KarstenTWolf** [FakeFishGames/Barotrauma#4965] [Ideas and suggestions]
Adding the ability to externally adjust a delay's delay time would be a helpful addition without any actual 'cost'. Would allow for some dynamically timed or remote adjustable delay settings. Also severely uncomplicates dynamic delaying of non-binary signals.

One decision would be to have a setting on what happens if there's a signal 'inside' the delay when the delay changes. I propose either treating the delay value change the same as reset when a different signal received. Having the signal 'complete' at the old or new delay value could also be an optional toggle.

Personally I've been trying to solve for this without, and it's... a learning experience. You can treat it as a countdown problem, but then you run into having to decide a countdown interval (seconds are great and all but what if I want 1.5second delays). Using an OSC to get a fully dynamic 'delay' time works but then solving for the 'reset when signal received' becomes a massive complicating factor.

**Comment by Xeavas** [comment]
Possibly having it called `set_delay` would also work but otherwise the idea sounds good.
